server/routes/offenceCodeDecisions/offenceCodeDecisions.ts

```diff
@@ -4,72 +4,90 @@ import decisionTree from '../../offenceCodeDecisions/DecisionTree'
-    const draftAdjudication = await this.placeOnReportService.getDraftAdjudicationDetails(
-      Number(adjudicationNumber),
-      user
-    )
-    const prisonerDetails = await this.placeOnReportService.getPrisonerDetails(
-      draftAdjudication.draftAdjudication.prisonerNumber,
-      user
-    )
-    return {
-      offenderFirstName: properCaseName(prisonerDetails.firstName),
-      offenderLastName: properCaseName(prisonerDetails.lastName),
-    }
-  }
+  constructor(private readonly placeOnReportService: PlaceOnReportService, private readonly userService: UserService) {}
 
   private renderView = async (req: Request, res: Response, pageData?: PageData): Promise<void> => {
     const { adjudicationNumber, incidentRole } = req.params
-    const { error, selectedDecisionId } = pageData
-    const placeholderValues = await this.placeholderValues(adjudicationNumber, res)
-    const path = req.path.replace(`/${adjudicationNumber}/${incidentRole}/`, '')
-    const decision = decisionTree.findByUrl(path)
-
+    const { errors } = pageData
+    const { user } = res.locals
+    const decisionForm = pageData // The form backing this page
+    const placeholderValues = await this.placeholderValues(adjudicationNumber, user)
+    const decision = decisionTree.findByUrl(req.path.replace(`/${adjudicationNumber}/${incidentRole}/`, ''))
     const pageTitle = decision.getTitle().getProcessedText(placeholderValues, incidentRole as IncidentRole)
     const questions = decision.getChildren().map(d => {
       return {
         id: d.id(),
         label: d.getQuestion().getProcessedText(placeholderValues),
+        type: d.getType().toString(),
       }
     })
-    return res.render(`pages/${decision.getPage() || 'offenceCodeDecisions'}`, {
-      errors: error ? [error] : [],
-      selectedDecisionId,
+    const selectedDecisionViewData = await this.viewDataFromDecisionForm(decisionForm, user)
+    return res.render(`pages/offenceCodeDecisions`, {
+      errors: errors || [],
+      decisionForm,
+      selectedDecisionViewData,
       questions,
       pageTitle,
       pageData,
     })
   }
 
-  view = async (req: Request, res: Response): Promise<void> => this.renderView(req, res, {})
+  view = async (req: Request, res: Response): Promise<void> => {
+    if (req.query.selectedPerson) {
+      // We are coming back from a user selection. We want to record this in the DecisionForm stored on the session and
+      // then redirect to the view page after removing the request parameter.
+      updateSessionDecisionForm(req, req.query.selectedPerson as string)
+      return res.redirect(
+        url.format({
+          pathname: this.urlHere(req),
+        })
+      )
+    }
+    // We are viewing this page. If we have come from a user selection then we should have the previous state of the
+    // form in the session, so we render that now and remove it from the session.
+    return this.renderView(req, res, getAndDeleteSessionDecisionForm(req) || {})
```

Comment:
On a view, once we have read everything we need to from the session we remove it. This means that every time we view this page we can be sure that the session has been wiped.